src/sardana/macroserver/scan/gscan.py

```diff
@@ -2294,14 +2300,19 @@ def _go_through_waypoints(self):
                 self.macro.debug(
                     "Moving to waypoint position: %s" % repr(final_pos))
                 motion.move(final_pos)
-            finally:
+            except Exception as e:
+                msg = "Motion did not start properly.\n{0}".format(e)
+                self.debug(msg)
+                raise ScanException("move to final position failed")
+            else:
```

Comment:
Only wait for the measurement group to end (including the timeout) only if no exception was raised.